pkg/rancher-desktop/preload/extensions.ts

```diff
@@ -5,21 +5,266 @@
+      throw new TypeError(getTypeErrorMessage('options.cwd', 'string', options?.cwd));
+    }
+    if (typeof options?.env !== 'undefined') {
+      if (typeof options.env !== 'object') {
+        throw new TypeError(getTypeErrorMessage('options.env', 'object', options.env));
+      }
+      for (const [k, v] of Object.entries(options.env)) {
+        if (!['undefined', 'string'].includes(typeof v)) {
+          throw new TypeError(getTypeErrorMessage(`options.env.${ k }`, 'string', v));
+        }
+      }
+    }
+
+    const execId = `${ scope }-${ nextId++ }`;
+    // Build options to pass to the main process, while not trusting the input
+    // too much.
+    const safeOptions: SpawnOptions = {
+      command: [`${ cmd }`].concat(Array.from(args).map(arg => `${ arg }`)),
+      execId,
+      scope,
+      ...(typeof options?.cwd === 'string' ? { cwd: `${ options.cwd }` } : {}),
+      ...(options?.env ? { env: Object.fromEntries(Object.entries(options.env).map(([k, v]) => [`${ k }`, `${ v }`])) } : {}),
+    };
+
+    if (options && isSpawnOptions(options)) {
+      // Build the object to return to the caller.  We manually define
+      // properties with symbol keys so they can't be enumerated (to avoid
+      // people accidentally clobbering our stuff).
+      const proc = Object.defineProperties({}, {
+        [id]: {
+          enumerable: false, value: execId, writable: false,
+        },
+        [stdout]: {
+          enumerable: false, value: '', writable: true,
+        },
+        [stderr]: {
+          enumerable: false, value: '', writable: true,
+        },
+        [stream]: {
+          enumerable: false, value: options.stream, writable: false,
+        },
+        close: {
+          value: function() {
+            ipcRenderer.send('extensions/spawn/kill', execId);
+            delete outstandingProcesses[execId];
+          },
+        },
+      }) as execProcess & v1.ExecProcess;
+
+      outstandingProcesses[execId] = new WeakRef(proc);
+      ipcRenderer.send('extensions/spawn/streaming', safeOptions);
+
+      return proc;
+    }
+
+    return (async() => {
+      const response = await ipcRenderer.invoke('extensions/spawn/blocking', safeOptions);
+
+      console.debug(`spawn/blocking got result:`, response);
+
+      return {
+        cmd:    response.cmd,
+        signal: typeof response.result === 'string' ? response.result : undefined,
+        code:   typeof response.result === 'number' ? response.result : undefined,
+        stdout: response.stdout,
+        stderr: response.stderr,
+        lines() {
+          return response.stdout.split(/\r?\n/);
+        },
+        parseJsonLines() {
+          return response.stdout.split(/\r?\n/).filter(line => line).map(line => JSON.parse(line));
+        },
+        parseJsonObject() {
+          return JSON.parse(response.stdout);
+        },
+      };
+    })();
+  }
+
+  return exec;
+}
+
+function getProcess(id: string): execProcess | undefined {
```

Comment:
No, as above — when we get it we don't care about the `v1.ExecProcess` interface. (Notably, I don't trust the object enough to call `close` on it, because somebody might figure out how to clobber the method. That's why I'm using symbol properties that are non-enumerable, so the user can't even see them.)